sync15-adapter/src/request.rs

```diff
@@ -0,0 +1,1203 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use util::ServerTimestamp;
+use bso_record::{BsoRecord, EncryptedPayload};
+
+use serde_json;
+use std::fmt;
+use std::collections::HashMap;
+use std::default::Default;
+use url::{Url, UrlQuery, form_urlencoded::Serializer};
+use error::{self, Result};
+use hyper::{StatusCode};
+use reqwest::Response;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum RequestOrder { Oldest, Newest, Index }
+
+header! { (XIfUnmodifiedSince, "X-If-Unmodified-Since") => [ServerTimestamp] }
+header! { (XLastModified, "X-Last-Modified") => [ServerTimestamp] }
+header! { (XWeaveTimestamp, "X-Weave-Timestamp") => [ServerTimestamp] }
+
+impl fmt::Display for RequestOrder {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            &RequestOrder::Oldest => f.write_str("oldest"),
+            &RequestOrder::Newest => f.write_str("newest"),
+            &RequestOrder::Index => f.write_str("index")
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
```

Comment:
a good candidate (probably among others) for #[derive(Default)]

Reply:
I didn't do this because it's a bad idea to create one of these without specifying a collection.